src/myrtlespeech/data/dataset/composite.py

```diff
@@ -0,0 +1,73 @@
+from typing import Tuple
+
+import torch
+from torch.utils.data import Dataset
+
+
+class Composite(Dataset):
+    """Combines multiple individual datasets.
+
+    An Iterable of :py:class:`torch.utils.data.Dataset`s are combined to
+    create a single :py:class:`torch.utils.data.Dataset`.
+
+    Args:
+        *children: the individual datasets that are combined to create this
+            composite dataset.
+    """
+
+    def __init__(self, *children: Dataset):
+        self.children = children
+        self.durations = [
+            dur for child in self.children for dur in child.durations
+        ]
+        self.child_map = [
+            (j, i)
+            for j, child in enumerate(self.children)
+            for i in range(len(child.durations))
+        ]
+        self._sort_by_duration()
+
+    def __getitem__(self, index: int) -> Tuple[torch.Tensor, str]:
+        r"""Returns the sample at ``index`` in the dataset.
+
+        The samples are in ascending order by *untransformed* audio sample
+        length.
+
+        Untransformed samples are :py:class:`torch.Tensor` with size ``(1,
+        samples)``.
+
+        Args:
+            index: Index.
+
+        Returns:
+            Tuple of ``(audiodata, target)`` where audiodata is the possibly
+            transformed audio sample and target is the possibly transformed
+            target transcription.
+        """
+        child_id, child_index = self.child_map[index]
+        return self.children[child_id][child_index]
+
+    def __len__(self) -> int:
+        return len(self.durations)
+
+    def _sort_by_duration(self) -> None:
+        """Orders the loaded data by audio duration, shortest first."""
+        total_samples = len(self.durations)
+        if total_samples == 0:
+            return
+        samples = zip(self.child_map, self.durations)
+        sorted_samples = sorted(samples, key=lambda sample: sample[1])
+        self.child_map, self.durations = [
+            list(c) for c in zip(*sorted_samples)
+        ]
+        assert (
+            total_samples == len(self.child_map) == len(self.durations)
+        ), "_sort_by_duration len mis-match"
+
+    def __repr__(self) -> str:
+        return (
+            self.__class__.__name__
+            + "["
+            + ",".join([child.__repr__() for child in self.children])
+            + "]"
```

Comment:
Typically `__repr__` uses parentheses as:

> If at all possible, this should look like a valid Python expression that could be used to recreate an object with the same value (given an appropriate environment).

- https://docs.python.org/3/reference/datamodel.html#object.__repr__

Are braces being used here for a reason?

Reply:
I suspect @SamG97 was using braces to demonstrate that there is a collection of child datasets. 

I've kept this idea but updated to the following so that it looks like a python expression:
```
            + "(datasets=["
            + ",".join([child.__repr__() for child in self.children])
            + "])"
```